src/tools.ts

```diff
@@ -27,718 +32,146 @@ class ToolException extends Error {
-            if (lowercaseValue === 'true') {
-              result[key] = true;
-            } else if (lowercaseValue === 'false') {
-              result[key] = false;
-            } else {
-              result[key] = Boolean(part);
-            }
-          } else if (expectedType === 'array') {
-            // If we expect an array but have a single value, wrap it in an array
-            result[key] = [part];
-          } else {
-            result[key] = part;
-          }
-        }
-      });
-
-      logger.debug('Parsed comma-separated values to:', result);
-      return result;
-    }
-  } catch (e) {
-    logger.debug('Error processing comma-separated values:', e);
-  }
-
-  // Fallback for single value inputs
-  if (schemaProperties.length > 0) {
-    try {
-      const stringValue = String(input).trim();
-      const numValue = Number(stringValue);
-
-      // For single value inputs, map to the first schema property
-      if (!isNaN(numValue)) {
-        const result: Record<string, any> = {};
-        result[schemaProperties[0]] = numValue;
-        logger.debug('Created object from single number value:', result);
-        return result;
-      }
-
-      // Special case: If we expect parameters 'a' and 'b', create default object
-      if (schemaProperties.includes('a') && schemaProperties.includes('b')) {
-        // Extract numbers if possible
-        const numMatches = stringValue.match(/\d+/g);
-        if (numMatches && numMatches.length >= 2) {
-          const result: Record<string, any> = {
-            a: Number(numMatches[0]),
-            b: Number(numMatches[1]),
-          };
-          logger.debug('Created a/b object from string with numbers:', result);
-          return result;
-        }
-      }
-    } catch (e) {
-      logger.debug('Error in fallback processing:', e);
-    }
-  }
-
-  // Return a last-resort empty object that matches the schema
-  const fallback: Record<string, any> = {};
-  schemaProperties.forEach(prop => {
-    const expectedType = schemaTypes[prop] || 'string';
-    if (expectedType === 'array') {
-      fallback[prop] = [];
-    } else if (expectedType === 'object') {
-      fallback[prop] = {};
-    } else if (expectedType === 'number') {
-      fallback[prop] = 0;
-    } else if (expectedType === 'boolean') {
-      fallback[prop] = false;
-    } else {
-      fallback[prop] = '';
-    }
-  });
-  logger.debug('Returning fallback object with proper types:', fallback);
-  return fallback;
-}
+export type ResponseFormat = 'text' | 'content_and_artifact';
 
 /**
  * Convert an MCP tool to a LangChain tool.
  *
  * @param client - The MCP client
- * @param toolName - The name of the tool
- * @param toolDescription - The description of the tool
- * @param toolSchema - The schema of the tool
- * @returns A LangChain tool implementing StructuredToolInterface
+ * @param tool - The MCP tool to convert
+ * @param toolSchema - Tool schema (unused in simplified implementation)
```

Comment:
do we need this? or is this kept for backwards compatibility? if so, can we indicate that somehow?

Reply:
Changed the comment